Automattic/pocket-casts-android
# [Auto Downloading] - Remove Auto Download Limit Setting from Podcast

## Description
- Removes the Auto download limit from podcast setting and keeping only global auto download limit setting
- See the doc for context: p1728629479247849-slack-C07QMC2GB1B


Fixes #3014

## Testing Instructions

### Test DB migration
1. Checkout `main` branch
2. Install the App
3. Subscribe to some podcast 
4. Open the podcast setting
5. ✅ Have auto download enabled
6. Checkout to this branch
7. Subscribe to some podcast and have auto downloaded enabled for it
8. ✅ Ensure you don't see any crash  


## Checklist
